runtime/seccomp_filter.c

```diff
@@ -0,0 +1,135 @@
+
+#define _GNU_SOURCE
+#include <fcntl.h>
+#include <linux/filter.h>
+#include <linux/seccomp.h>
+#include <linux/unistd.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <sys/prctl.h>
+#include <sys/syscall.h>
+#include <unistd.h>
+
+// bpf filter to forbid the seccomp syscall
+struct sock_filter forbid_seccomp_filter[] = {
+    // load the syscall number
+    BPF_STMT(BPF_LD | BPF_W | BPF_ABS, (offsetof(struct seccomp_data, nr))),
+    // compare syscall number to seccomp() and jump 0 ahead if equal, 1 ahead if
+    // not
+    BPF_JUMP(BPF_JMP | BPF_JEQ | BPF_K, __NR_seccomp, 0, 1),
+    BPF_STMT(BPF_RET | BPF_K, SECCOMP_RET_KILL),
+    BPF_STMT(BPF_RET | BPF_K, SECCOMP_RET_ALLOW),
+};
+
+struct sock_fprog forbid_seccomp_prog = {
+    .len = (unsigned short)(sizeof(forbid_seccomp_filter) /
+                            sizeof(forbid_seccomp_filter[0])),
+    .filter = forbid_seccomp_filter,
+};
+
+// shorthand for an equality comparison jump of 0 (eq) or 1 (neq) followed by
+// a return of the given policy (used in the equal case)
+#define BPF_SYSCALL_POLICY(name, policy)                                       \
+  BPF_JUMP(BPF_JMP | BPF_JEQ | BPF_K, __NR_##name, 0, 1),                      \
+      BPF_STMT(BPF_RET | BPF_K, SECCOMP_RET_##policy)
+
+long syscall(long no, ...);
+
+int configure_seccomp(void) {
+  struct sock_filter filter[] = {
+      BPF_STMT(BPF_LD | BPF_W | BPF_ABS, (offsetof(struct seccomp_data, nr))),
+      BPF_SYSCALL_POLICY(write, ALLOW),
+      // this would compare syscall number to write() and allow if it matches
+      /*BPF_JUMP(BPF_JMP | BPF_JEQ | BPF_K, __NR_write, 0, 1),
+      BPF_STMT(BPF_RET | BPF_K, SECCOMP_RET_ALLOW),*/
```

Comment:
I don't think these comment are necessary, the macros is pretty straightforward.